bindings/python/pinocchio/visualize/viser_visualizer.py

```diff
@@ -0,0 +1,290 @@
+        elif isinstance(geom, MESH_TYPES):
+            mesh = trimesh.load(geometry_object.meshPath)
+            if color is None:
+                frame = self.viewer.scene.add_mesh_trimesh(name, mesh)
+            else:
+                frame = self.viewer.scene.add_mesh_simple(
+                    name,
+                    mesh.vertices,
+                    mesh.faces,
+                    color=color_override[:3],
+                    opacity=color_override[3],
+                )
+        else:
+            raise RuntimeError(f"Unsupported geometry type for {name}: {type(geom)}")
+
+        self.frames[name] = frame
+
+    def display(self, q=None):
+        """
+        Display the robot at configuration q in the viewer by placing all the bodies
+        """
+        if q is not None:
+            pin.forwardKinematics(self.model, self.data, q)
+
+        if self.collisionRootFrame.visible:
+            self.updatePlacements(pin.GeometryType.COLLISION)
+
+        if self.visualRootFrame.visible:
+            self.updatePlacements(pin.GeometryType.VISUAL)
+
+        if self.framesRootFrame.visible:
+            self.updateFrames()
+
+    def displayCollisions(self, visibility):
+        self.collisionRootFrame.visible = visibility
+        self.updatePlacements(pin.GeometryType.COLLISION)
+
+    def displayVisuals(self, visibility):
+        self.visualRootFrame.visible = visibility
+        self.updatePlacements(pin.GeometryType.VISUAL)
+
+    def displayFrames(self, visibility):
+        self.framesRootFrame.visible = visibility
+        self.updateFrames()
+
+    def drawFrameVelocities(self, *args, **kwargs):
+        raise NotImplementedError("drawFrameVelocities is not yet implemented.")
+
+    def updatePlacements(self, geometry_type):
+        if geometry_type == pin.GeometryType.VISUAL:
+            geom_model = self.visual_model
+            geom_data = self.visual_data
+            prefix = self.viewerRootNodeName + "/visual"
+        else:
+            geom_model = self.collision_model
+            geom_data = self.collision_data
+            prefix = self.viewerRootNodeName + "/collision"
+
+        pin.updateGeometryPlacements(self.model, self.data, geom_model, geom_data)
+        for geometry_object in geom_model.geometryObjects:
+            # Get mesh pose.
+            M = geom_data.oMg[geom_model.getGeometryId(geometry_object.name)]
+
+            # Update viewer configuration.
+            frame_name = prefix + "/" + geometry_object.name
+            frame = self.frames[frame_name]
+            frame.position = M.translation * geometry_object.meshScale
+            frame.wxyz = pin.Quaternion(M.rotation).coeffs()[
+                [3, 0, 1, 2]
+            ]  # Pinocchio uses xyzw
+
+    def updateFrames(self):
+        pin.updateFramePlacements(self.model, self.data)
+        for frame in self.model.frames:
+            # Get frame pose.
+            M = self.data.oMf[self.model.getFrameId(frame.name)]
```

Comment:
Prefer to iterate as
```python
for i, frame in enumerate(self.model.frames):
```
to avoid querying the frame ID by its name and occurring a lookup for every frame for every call to this function.

Reply:
Oh wow, this whole time I was under the impression that there was an underlying map from frame name to ID 😰 -- thanks for sharing! I may have some other code to go fix